src/Symfony/Component/HttpFoundation/Session/Storage/Handler/RedisSessionHandler.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+/**
+ * Redis based session storage handler based on the Redis class
+ * provided by the PHP redis extension.
+ *
+ * Direct port of MemcachedSessionHandler to Redis.
+ *
+ * @see http://php.net/redis
+ *
+ * @author Drak <drak@zikula.org>
+ * @author Dalibor Karlović <dkarlovi@gmail.com>
+ */
+class RedisSessionHandler extends AbstractSessionHandler
+{
+    /**
+     * @var \Redis
+     */
+    private $redis;
+
+    /**
+     * @var int Time to live in seconds
+     */
+    private $ttl;
+
+    /**
+     * @var string Key prefix for shared environments
+     */
+    private $prefix;
+
+    /**
+     * List of available options:
+     *  * prefix: The prefix to use for the Redis keys in order to avoid collision
+     *  * expiretime: The time to live in seconds.
+     *
+     * @param \Redis $redis   A \Redis instance
+     * @param array  $options An associative array of Redis options
+     *
+     * @throws \InvalidArgumentException When unsupported options are passed
+     */
+    public function __construct(\Redis $redis, array $options = null)
```

Comment:
I think this should also support Predis, as the Cache does. There's little effort required to support both drivers.

Reply:
Sure, I'll add it.

Reply:
@nicolas-grekas do you mean stuff from `RedisTrait`? It looks rather complex and I (guess I?) can't use it as it's from another component. Would I need to replicate / inline the whole thing?

Reply:
Not sure you need to borrow anything, I was just pointing at this to show we already managed to have a single class able to deal with both drivers, so we should be able to make it again here :)

Reply:
Would we also want to support Array and Cluster too?

Reply:
likely yes :)

Reply:
Oh. :)